service/resource/legacyv2/resource.go

```diff
@@ -235,6 +232,17 @@ func (r *Resource) newDeleteChange(ctx context.Context, obj, currentState, desir
 		}
 	}
 
+	// Create a service account for the cleanup job.
```

Comment:
AIUI this service account will remain after the deletion, is that ok, shouldn't it be deleted too? maybe it is cleaned up with the destroyer namespace pls correct me if I'm wrong.

Reply:
Yes the service account is namespaced, so when you delete it, it will disappear (Already tested :) )